sqlglot/dialects/hive.py

```diff
@@ -664,3 +663,22 @@ def struct_sql(self, expression: exp.Struct) -> str:
                     values.append(e)
 
             return self.func("STRUCT", *values)
+
+        def alterset_sql(self, expression: exp.AlterSet) -> str:
+            exprs = self.expressions(expression, flat=True)
+            location = self.sql(expression, "location")
+            location = f" LOCATION {location}" if location else ""
+            file_format = self.expressions(expression, key="file_format", flat=True, sep=" ")
+            file_format = f" FILEFORMAT {file_format}" if file_format else ""
+            serde = self.sql(expression, "serde")
+            serde = f" SERDE {serde}" if serde else ""
+            tags = self.expressions(expression, key="tag", flat=True, sep="")
+            tags = f" TAGS {tags}" if tags else ""
+
+            return f"SET{serde}{exprs}{location}{file_format}{tags}"
+
+        def serdeproperties_sql(self, expression: exp.SerdeProperties) -> str:
+            prefix = "WITH " if expression.args.get("with") else ""
+            exprs = self.expressions(expression, flat=True, sep=", ")
+
+            return f"{prefix}SERDEPROPERTIES ({exprs})"
```

Comment:
I think not, because e.g. Spark/Databricks support the following statement:

```
ALTER TABLE table_identifier [ partition_spec] SET SERDEPROPERTIES ( key1 = val1, key2 = val2, ... )
```

This would be parsed as `exp.SerdeProperties` which is wanted, but if it's always generated back as `SET WITH SERDEPROPERTIES` that doesn't conform to the specs, right?

Reply:
Got it, I missed that context. Sounds good.